security/pkg/nodeagent/cache/secretcache.go

```diff
@@ -570,33 +570,34 @@ func (sc *SecretCache) rotate(updateRootFlag bool) {
 			return true
 		}
 
-		if sc.configOptions.CredFetcher != nil {
-			// Refresh token through credential fetcher.
-			cacheLog.Debugf("%s getting a new token through credential fetcher", logPrefix)
-			t, err := sc.configOptions.CredFetcher.GetPlatformCredential()
-			if err != nil {
-				cacheLog.Warnf("%s credential fetcher failed to get a new token, continue using the original token: %v", logPrefix, err)
-			} else {
-				secret.Token = t
-			}
-		}
-
 		// Re-generate secret if it's expired.
 		if sc.shouldRotate(&secret) {
 			atomic.AddUint64(&sc.secretChangedCount, 1)
-			// Send the notification to close the stream if token is expired, so that client could re-connect with a new token.
-			if sc.isTokenExpired(&secret) && !sc.useCertToRotate() {
-				cacheLog.Debugf("%s token expired", logPrefix)
-				// TODO(myidpt): Optimization needed. When using local JWT, server should directly push the new secret instead of
-				// requiring the client to send another SDS request.
-				sc.callbackWithTimeout(connKey, nil /*nil indicates close the streaming connection to proxy*/)
-				return true
-			}
 
+			// TODO: not clear why a wg is used, and then a wait - instead of just running the code. Cleanup ?
```

Comment:
1 per cert, so we regen in parallel? Not sure its actually a good idea, but the WG is not a no-op

Reply:
In practice - there is only one cert we renew, the workload cert. 

The code probably meant something in the old nodeagent.  For the SDS for Gateway - I don't think we're reusing any of this.

Reply:
Yes, this was introduced in the old node agent, which served multiple pods.
https://github.com/istio/istio/pull/9055/files

Reply:
Can we remove the wg?

Reply:
In a separate PR - but it may not be needed, once we have the Pilot XDS server running by default in agent
we'll just convert this to a Generator.